specs-go/config.go

```diff
@@ -58,8 +58,80 @@ type Process struct {
 	OOMScoreAdj *int `json:"oomScoreAdj,omitempty" platform:"linux"`
 	// SelinuxLabel specifies the selinux context that the container process is run as.
 	SelinuxLabel string `json:"selinuxLabel,omitempty" platform:"linux"`
+	// Landlock specifies the Landlock unprivileged access control settings for the container process.
+	Landlock Landlock `json:"landlock,omitempty" platform:"linux"`
 }
 
+// Landlock specifies the Landlock unprivileged access control settings for the container process.
+type Landlock struct {
+	// Ruleset identifies a set of rules (i.e., actions on objects) that need to be handled.
+	Ruleset LandlockRuleset `json:"ruleset,omitempty" platform:"linux"`
+	// Rules are the security policies (i.e., actions allowed on objects) to be added to an existing ruleset.
+	Rules []LandlockRule `json:"rules,omitempty" platform:"linux"`
+	// ABI is the specific Landlock ABI version.
+	ABI LandlockABIVersion `json:"abi,omitempty" platform:"linux"`
+}
+
+// LandlockRuleset identifies a set of rules (i.e., actions on objects) that need to be handled.
+type LandlockRuleset struct {
+	// HandledAccessFS is a list of actions that is handled by this ruleset and should then be
+	// forbidden if no rule explicitly allow them.
+	HandledAccessFS []LandlockFSAction `json:"handledAcessFS,omitempty" platform:"linux"`
+}
+
+// LandlockRule represents the security policies (i.e., actions allowed on objects) .
+type LandlockRule struct {
+	// Type is the Landlock rule type pointing to the rules to be added to an existing ruleset.
+	Type LandlockRuleType `json:"type,omitempty" platform:"linux"`
```

Comment:
Do we really need multiple `LandlockRule`s of the same type?

For example, we can do instead:
```
"rules": {
  "path_beneath": [
    {
      "allowedAccess": ["xxx"],
      "paths": ["xxx"]
    },
    {
      "allowedAccess": ["xxx"],
      "paths": ["xxx"]
    }
  ]
}
```

Reply:
I agree that it would make the configuration lighter.

Reply:
Thanks for the feedbacks! Use an object with rule-type keys instead to make it lighter.